packages/foundry/contracts/other/notional/NotionalJoin.sol

```diff
@@ -181,26 +144,24 @@ contract NotionalJoin is IJoin, ERC1155TokenReceiver, AccessControl() {
         uint256 underlyingBalance = IERC20(underlying).balanceOf(address(this));
         uint256 storedBalance_ = storedBalance;
         accrual = underlyingBalance.wdiv(storedBalance_); // There is a rounding loss here. Some wei will be forever locked in the join.
-        // This becomes now to an exit-only underlying Join.
-        storedBalance = underlyingBalance;
+        
+        // transfer underlying to main join
+        IJoin(underlyingJoin).join(address(this), underlyingBalance.u128());
```

Comment:
You need to transfer the underlying to the join first.

Reply:
Wouldn't that be achieved by calling `join(address(this), underlyingBalance.u128())` ? 
My understanding is that it would move the underlying from the NotionalJoin to itself (underlyingJoin).

Reply:
Oh, I understand this now. 

Njoin would `transfer` the underlying to the main join then call `join`.

This is because the join checks the difference btw its own logged balance versus balanceOf (`available`), and moves only the shortfall (`amount - available`) to itself. 

```solidity
        uint256 available = token.balanceOf(address(this)) - _storedBalance; // Fine to panic if this underflows
        unchecked {
            storedBalance = _storedBalance + amount; // Unlikely that a uint128 added to the stored balance will make it overflow
            if (available < amount) token.safeTransferFrom(user, address(this), amount - available);
        }
```

This clicked while I watching your Yield v2 Vault Walkthrough on youtube. Thanks!